### PR TITLE
Cut 0.34.0

### PR DESCRIPTION
**Pull request by s-urbaniak**
This cuts a new prometheus rules. Any help is appreciated if I'm doing this right :-)

Changes from the previous release: https://github.com/coreos/prometheus-operator/compare/release-0.33...25144969947563ccc480079b2533375a84d84adf

/cc @LiliC @brancz @metalmatze

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,34 @@
...
+* [FEATURE] Add [prometheus,alertmanager]-instance-namespaces cmdline parameter (#2783)
+* [FEATURE] Add configSecret element to the AlertmanagerSpec (#2827)
+* [FEATURE] Add enforcedNamespaceLabel to Prometheus CRD (#2820)
+* [FEATURE] Add exec probes against localhost:9090 to Prometheus if listenLocal is set to true (#2763)
+* [FEATURE] Add honorTimestamps field to Prometheus, Podmonitor, and ServiceMonitor CRD (#2800)
+* [FEATURE] Add ignoreNamespaceSelectors field to Prometheus CRD (#2816)
+* [FEATURE] Add local configuration options in jsonnet/prometheus-operator (#2794)
+* [FEATURE] Add overrideHonorLabels to Prometheus CRD (#2806)
+* [FEATURE] Reference secrets instead of local files (#2716)
+* [ENHANCEMENT] Add basic podmonitor docs (#2807)
```
Comment (metalmatze) [inline CHANGELOG.md]: We usually left these out.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+* [ENHANCEMENT] Add missing json struct tag for ArbitraryFSAccessThroughSMsConfig (#2808)
+* [ENHANCEMENT] Add testing global settings (#2811)
+* [ENHANCEMENT] Allow running only part of e2e tests and use it in CI (#2804)
+* [ENHANCEMENT] Bump jsonnet and golangci-lint versions in (#2784)
+* [ENHANCEMENT] Clean up travis yaml (#2765)
```
Comment (metalmatze) [inline CHANGELOG.md]: That's not really a user facing enhancement, thus we leave this usually out

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+* [ENHANCEMENT] Ensure containers have "FallbackToLogsOnError" termination policy (#2819)
+* [ENHANCEMENT] Fix a typo in documentation (#2760)
```
Comment (metalmatze) [inline CHANGELOG.md]: Not user facing either, let's ignore it for the change log

**Comment by s-urbaniak** [comment]
@metalmatze thanks a lot for the quick review! :-) addressed comments, removed non user facing changelog entries, PTAL

**Pull request merged**